gallery_dl/extractor/generic.py

```diff
@@ -26,12 +26,34 @@ class GenericExtractor(Extractor):
     # Based on: https://tools.ietf.org/html/rfc3986#appendix-B
     pattern += r"""
         (?P<scheme>https?://)?          # optional http(s) scheme
-        (?P<domain>[-\w\.]+)            # required domain
+        (?P<domain>[^/?#]+)             # required domain
```

Comment:
As far as I can tell, this doesn't actually do anything relating to IDNs. 
It only allows special characters.

`\w` already matches any Unicode word character.
Your added tests already pass without this change.

```
\w

    For Unicode (str) patterns:

        Matches Unicode word characters;
        this includes alphanumeric characters (as defined by str.isalnum() as well as the underscore (_)
```

Reply:
Thank you for pointing this out. I've always thought `\w` was equivalent to `[A-Za-z0-_]`. What's interesting is that `[^/?#]` is used by https://www.rfc-editor.org/rfc/rfc3986#appendix-B as well, but I guess it's better to use `[-\w\.]` for domain names since special characters are not allowed anyway.

I'm going to mark this PR as WIP for now, seeing as this commit is a side product and doesn't do anything besides adding a few tests.